### ISSUE TITLE
LSP function parameters completions/signature sugestions not working on neovim nightly

### ISSUE THREAD

**Issue by KayJay7** [nvim-lua/completion-nvim#372]
Just like the title says, when writing a function it should pop up signature suggestions, but it gives an error instead. More details below. Everything else seems to be working.

**My testing minimal init.vim**
```vim-script
call plug#begin('~/.vim/plugged') 
    Plug 'neovim/nvim-lspconfig'
    Plug 'nvim-lua/completion-nvim'
call plug#end()
lua require'lspconfig'.clangd.setup{on_attach=require'completion'.on_attach}
au Filetype lua setl omnifunc=v:lua.vim.lsp.omnifunc
" Use <Tab> and <S-Tab> to navigate through popup menu
inoremap <expr> <Tab>   pumvisible() ? "\<C-n>" : "\<Tab>"
inoremap <expr> <S-Tab> pumvisible() ? "\<C-p>" : "\<S-Tab>"

" Set completeopt to have a better completion experience
set completeopt=menuone,noinsert,noselect
```

**How to reproduce**
Open a C or C++ files (happens with other LSPs too, I've tried it with rust-analyzer). Try to call a function. Every time you start writing an argument, it will show an error message on the status line. The error message is:
```
focusable_preview is deprecated. Please use open_floating_preview and pass focus_id = [unique_name] instead
Error executing vim.schedule lua callback: /usr/share/nvim/runtime/lua/vim/lsp/util.lua:1322: syntax: expected string, got table
stack traceback:
        /usr/share/nvim/runtime/lua/vim/lsp/util.lua:1322: in function 'focusable_preview'
        ...lugged/completion-nvim/lua/completion/signature_help.lua:45: in function 'handler'
        /usr/share/nvim/runtime/lua/vim/lsp.lua:896: in function 'cb'
        vim.lua:277: in function <vim.lua:277>
```

**Expected behavior**
It should open a floating buffer with signature suggestions for the function, showing parameters name and eventually other infos provided by the LSP. I think it should even suggest variables to use as argument, but maybe it depends on the LSP, I'm not sure.

So far, I've tried what the worning message suggested: replacing the lines at `signature_help.lua:45-49` with the following:
```lua
-- line 44, unchanged ...
      lines = vim.lsp.util.trim_empty_lines(lines)
      local synt = vim.lsp.util.try_trim_markdown_code_blocks(lines)
      local bufnr, _ = vim.lsp.util.open_floating_preview(lines, synt, {focus_id = method})
-- line 48, previously line 50, unchanged...
```
Which is basically what the `focusable_preview` function at `util.lua:1034` does, minus the call to the callback function.
It did something, it shows a floating buffer instead of an error, I'm not sure if the buffer shows what it's supposed to (I don't remember for sure), but it does open; anyway, it doesn't work properly, tabbing should select the options in the buffer, but it just tabs the text or whatever it's mapped in the init.vim file (in my actual init.vim, it should open the suggestion buffer if not open, instead it opens a new suggestion buffer which does select properly, but it doesn't use the information from the LSP). Surprisingly enough, rust-analyzer seems to kind of work after this changes, but I haven't properly checked, and clangd still doesn't work.

Nvim, vim-plug, and pretty much everything else, is up to date. Nvim is the nightly version (other plugins require version 0.5.0).

**Comment by ranjithshegde** [comment]
check with master now? I think #371 fixed this

**Comment by KayJay7** [comment]
Yes, it solved the problem, I guess the other behavior was already there, and I just didn't notice. I probably never used tab before writing at least the first letter of the argument.

I noticed another weird new behavior: when I use the completion on a rust function, instead of just writing the first parenthesis and displaying options, it writes down the entire function call, filling the arguments with their signature name. I'm not sure if it is even the fault of this plugin, the server, or just another plugin. I will look into it, but I guess it's for another issue, I'm closing this one.

**Comment by KayJay7** [comment]
Solved by #371
